src/testE2E/amazonq/framework/framework.ts

```diff
@@ -0,0 +1,101 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { JSDOM } from 'jsdom'
+/**
+ * JSDOM is used to help hoist MynahUI to running in a node environment vs in the browser (which is what it's made for)
+ */
+const dom = new JSDOM(undefined, {
```

Comment:
As a question, I'm not 100% sure.

I think unless you actually have usages, it will work in TS, but are you specifically intending for this to be a "side-effect of an import"?

We could extract this to a function to make this more explicit like calling "injectJSDomforMynah", I think it would be more maintanable

Reply:
The problem is that it uses some browser apis that aren't available in node. That's why we have to use jsdom

Reply:
yeah, I understand the jsdom part. My main question is about this being a side effect of an import (a.k.a. top level code) as opposed to being explicitly overriding global variables by calling a method.